File: DE_2/contestants/nguyen_duy_hung/bai01.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    ifstream inp("bai01.inp");
    ofstream out("bai01.out");
    long long a,m,n,b,i,j;
    inp >> a >> b >> n >> m;
    long long s=1e9;
    i=pow(a,n);
    j=pow(b,m);
    out << i%s+j%s;
    out.close();
    return 0;
}

File: DE_2/contestants/bach_do/bai01.py
with open("bai01.inp","r") as fi:
    a,b,n,m=map(int,fi.readline().split())
with open("bai01.out","w") as fo:
    x=a**n
    y=b**m
    z=x+y
    fo.write(str(z))
fi.close()
fo.close()